spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -2443,25 +2443,32 @@ private void invokeBatchOnMessageWithRecordsOrList(final ConsumerRecords<K, V> r
 				}
 			}
 			Object sample = startMicrometerSample();
-			try {
-				if (this.wantsFullRecords) {
-					Objects.requireNonNull(this.batchListener).onMessage(records, // NOSONAR
-							this.isAnyManualAck
-									? new ConsumerBatchAcknowledgment(records, recordList)
-									: null,
-							this.consumer);
+
+			// Handle individual record tracing for batch mode if enabled
+			if (this.containerProperties.isRecordObservationsInBatch() && this.observationEnabled) {
```

Comment:
Isn't `this.observationEnabled` already covering `this.containerProperties.isRecordObservationsInBatch()`?